### PR TITLE
Enhance module bootloading

### PR DESCRIPTION
**Pull request by khenderick**
* Fix a bunch of small issues
* Ignore unavailable modules
* Ignore virtual/emulated modules
* Don't rely on version call being available
* Don't skip modules < 3.1.x
* Better logging

### CODE REVIEW

**Comment by khenderick** [comment]
@openmotics-ci test gateway=bugfix/better-bootloading frontend=develop

**Comment by khenderick** [comment]
@openmotics-ci test gateway=bugfix/better-bootloading frontend=develop

File: src/modules_bootloader.py [hunk 1]
```diff
@@ -269,14 +283,13 @@ def bootload(master_communicator, address, ihex, crc, blocks, logger):
...
-    logger('Verifying firmware')
-    do_command(logger,
-               master_communicator,
-               create_bl_action(master_api.modules_get_version(),
-                                {'addr': address}),
-               success_code=255)
-
-    logger('New version: v {0}.{1}.{2}'.format(result['f1'], result['f2'], result['f3']))
+    logger('Checking the version')
+    result = do_command(logger,
```
Comment (svanscho) [inline src/modules_bootloader.py]: doc link to status codes?

Comment (khenderick) [reply src/modules_bootloader.py]: Docs added

**Comment by khenderick** [comment]
@openmotics-ci test gateway=bugfix/better-bootloading frontend=develop

**Pull request merged**